sklearn/decomposition/_nmf.py

```diff
@@ -1717,3 +1759,617 @@ def inverse_transform(self, W):
     def _n_features_out(self):
         """Number of transformed output features."""
         return self.components_.shape[0]
+
+
+class MiniBatchNMF(NMF):
+    """Mini-Batch Non-Negative Matrix Factorization (NMF).
+
+    .. versionadded:: 1.1
+
+    Find two non-negative matrices (W, H) whose product approximates the non-
+    negative matrix X. This factorization can be used for example for
+    dimensionality reduction, source separation or topic extraction.
+
+    The objective function is:
+
+        .. math::
+
+            0.5 * ||X - WH||_{loss}^2
+
+            + alpha\\_W * l1_{ratio} * n\\_features * ||vec(W)||_1
+
+            + alpha\\_H * l1_{ratio} * n\\_samples * ||vec(H)||_1
+
+            + 0.5 * alpha\\_W * (1 - l1_{ratio}) * n\\_features * ||W||_{Fro}^2
+
+            + 0.5 * alpha\\_H * (1 - l1_{ratio}) * n\\_samples * ||H||_{Fro}^2
+
+    Where:
+
+    :math:`||A||_{Fro}^2 = \\sum_{i,j} A_{ij}^2` (Frobenius norm)
+
+    :math:`||vec(A)||_1 = \\sum_{i,j} abs(A_{ij})` (Elementwise L1 norm)
+
+    The generic norm :math:`||X - WH||_{loss}^2` may represent
+    the Frobenius norm or another supported beta-divergence loss.
+    The choice between options is controlled by the `beta_loss` parameter.
+
+    The objective function is minimized with an alternating minimization of W
+    and H.
```

Comment:
Nice layout.

The `l1_{ratio}` rendering is weird because math variables are not really meant to use multiple letters in LaTeX, though but I am not sure which notation to use instead.

Maybe something like:
```
q_{l_1}
```

Where we say in the in the legend that:

```
:math:`q_{l_1}`
```

corresponds to the `l1_ratio` parameter?

Reply:
> The l1_{ratio} rendering is weird because math variables are not really meant to use multiple letters in LaTeX, though but I am not sure which notation to use instead.

I do agree that the rendering can be improved (I forgot this parameter).

To me, there's a great (and probably underestimated) value in having consistency between mathematical notations (ideally from the original articles) and the parameters names exposed by the API.

How can we minimize this gap here?

To follow up with @vene's comment https://github.com/scikit-learn/scikit-learn/pull/16948#issuecomment-1006003641: I favor having rendered mathematical expressions in the doc at the code of a hardened source readability. My prior intuition is that people are generally reading the doc not the source code, and that the one would do generally are prone to be familiar with LaTeX. Let me know if I am wrong.

Stiil, I think we should be able to have both by find short names that are readable in source mode and that renders nicely.

Reply:
This is taken from the docstring of NMF.  If we want to improve it I think we should do it for both, but maybe in a dedicated PR

Reply:
Finally I pushed the changes for both here, it's simpler :)
I came up with this:
![nmf-loss-doc](https://user-images.githubusercontent.com/34657725/153164119-64b397dc-2347-43f4-85ae-80c7fae9c3ac.png)
I think it's a good compromise between readability in the doc and readability in the source. I'd rather keep the parameters written explicitly to have a direct match with the signature. I think it makes it much easier to understand the impact of each parameter at first sight.
Also I kept the explicit multiplications "*", which give a clearer separation between the params, especially when they are not single letter params.
Let me know what you think.